src/api/services/translations_cache.py

```diff
@@ -1,51 +1,140 @@
+            if isinstance(field_value, str):
+                return self.translate_text_field(context, field_value, target_language_code)
+            elif isinstance(field_value, dict):
+                return self.translate_json_field(context, field_value, target_language_code, ignoredKeys)
+            elif isinstance(field_value, list):
+                for item in field_value:
+                    if isinstance(item, dict):
+                        return self.translate_json_field(context, item, target_language_code, ignoredKeys)
+                    elif isinstance(item, str):
+                        return self.translate_text_field(context, item, target_language_code)
+                    else:
+                        continue
+            else:
+                continue
+
+    def translate_list_field (self, context: Context, value: list, target_language_code, ignoredKeys: Set) -> Tuple[ dict or None, None ]:
+        for item in value:
+            if isinstance(item, dict):
+                return self.translate_json_field(context, item, target_language_code, ignoredKeys)
+            elif isinstance(item, str):
+                return self.translate_text_field(context, item, target_language_code)
+            else:
+                continue
+
+    def translate_model (self, context: Context, args: dict) -> dict or None:
+        try:
+            model = args.get('model', None)
+            target_language_code = args.get('target_language_code', None)
+
+            if not model:
+                return None, CustomMassenergizeError("Please provide a valid model")
+
+            translation_meta = getattr(model, "TranslationMeta", None)
+
+            if not translation_meta:
+                return
+
+            translatable_fields = getattr(translation_meta, "fields_to_translate", None)
+
+            if len(translatable_fields) > 0:
+                records = model.objects.all()
+                for record in records:
+                    for field in translatable_fields:
+                        field_value = getattr(record, field)
+
+                        if isinstance(field_value, str):
+                            self.translate_text_field(context, field_value, target_language_code[0])
+                        elif isinstance(field_value, dict):
+                            self.translate_json_field(context, field_value, target_language_code[0], set())
+                        elif isinstance(field_value, list):
+                            for item in field_value:
+                                if isinstance(item, dict):
+                                    self.translate_json_field(context, item, set())
+                                elif isinstance(item, str):
+                                    self.translate_text_field(context, item, target_language_code)
+                                else:
+                                    continue
+                        else:
+                            continue
```

Comment:
I sent a PR here which i think might be helpful: https://github.com/massenergize/api/pull/1102

Basically, you want to go ahead and serialize like business as usual, then pass the serialized json to the utility i created (and you can set an exclusion list) and then it will take that json and call whatever underlying translation service and return you back another json.

Reply:
I'll look into how to incorporate it and also fix the packages issue.